internal/config/config.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"gopkg.in/yaml.v3"
+)
+
+// Config represents the complete application configuration
+type (
+	Log struct {
+		Level  string `yaml:"level"`
+		Format string `yaml:"format"`
+	}
+
+	Config struct {
+		Log Log `yaml:"log"`
+	}
+)
+
+const (
+	// Flags
+	LogLevelFlag  = "log.level"
+	LogFormatFlag = "log.format"
+)
+
+// DefaultConfig returns a Config with default values
+func DefaultConfig() *Config {
+	cfg := &Config{}
+
+	cfg.Log.Level = "info"
+	cfg.Log.Format = "text"
```

Comment:
suggestion: return a struct with default values
```
return &Config{
  Log: LogConfig{
    Level: "info",
    Format: "text",
  }
}
```

IMHO, makes it more clear what the default values are.

Reply:
or a config parsed from a default yaml file is also file.

Reply:
We can go with your suggestion for now but when the config grows quite long, we will have to revert to the current version for better readability.

Reply:
i am also asking from better readability point of view.

don't accept a suggestion "for now" , no point in accepting if you think it will change very soon.

Reply:
Resolving this, please reopen to discuss further.